src/Streamly/Coreutils/Uniq.hs

```diff
@@ -0,0 +1,214 @@
+import Control.Monad.Catch (MonadCatch, MonadThrow)
+import System.IO.Unsafe (unsafePerformIO)
+import Streamly.Internal.Data.SVar (MonadAsync)
+import Control.Monad.IO.Class (MonadIO, liftIO)
+import Streamly.Internal.Data.Stream.Serial(SerialT)
+import Streamly.Internal.Data.Stream.StreamK.Type (IsStream)
+
+-------------------------------------------------------------------------------
+-- Record for options used with uniq
+-------------------------------------------------------------------------------
+
+{-# INLINE intMax #-}
+
+intMax :: Int
+intMax = maxBound
+
+
+data UniqOptions = UniqOptions {
+                       count :: Bool
+                     , repeated :: Bool       -- display only duplicate lines once for each group
+                     , duplicate :: Bool      -- print all duplicate lines
+                     , skipFields :: Int      -- skips first skipFields number of non-empty strings
+                     , skipChar :: Int        -- skips first skipChar number of characters (after above op)
+                     , checkChar :: Int       -- take first checkChar number of characters for comparison (after above op)
+                     , ignoreCase :: Bool     -- whether to ignore case
+                     , unique :: Bool         -- print only unique lines
+                     , delimiter :: Char      -- to separate output lines
+                  }
+
+{-# INLINE defaultUniqOptions #-}
+
+defaultUniqOptions :: UniqOptions
+defaultUniqOptions = UniqOptions True False False 0 0 intMax False True '\n'
+
+
+-------------------------------------------------------------------------------
+-- helper functions for uniq
+-------------------------------------------------------------------------------
+
+
+-- convert stream of bytes to stream of characters - ignore case if bool is True
+
+ignCase :: (IsStream t, Monad m) => Bool -> t m Word8 -> t m Char
+ignCase flag strm = if flag == True
+                    then S.map toLower $ Un.decodeLatin1 strm
+                    else Un.decodeLatin1 strm
+
+
+-- convert stream of char to stream of Array Char by breaking on newline
+
+splitOnNewLine :: (MonadIO m, IsStream t, Monad m) => t m Char -> t m (A.Array Char)
+splitOnNewLine strm = S.splitOnSuffix (== '\n') A.write strm
+
+
+
+{-# INLINE safeHead #-}
+
+safeHead :: [a] -> Maybe a
+safeHead (x:_) = Just x
+safeHead [] = Nothing
+
+
+{-# INLINE eqCh #-}
+
+eqCh :: Char -> Char -> Bool
+eqCh s c = (isSpace s == True && isSpace c == True) || (isSpace s == False && isSpace c == False)
+
+
+{-# INLINE mergeStringChar #-}
+
+mergeStringChar :: String -> Char -> String
+mergeStringChar s c = case safeHead s of
+                                  Just l -> if eqCh l c == True
+                                            then s ++ [c]
+                                            else [c]
+                                  Nothing -> s ++ [c]
+
+
+
+
+skpField :: (IsStream t, Monad m) => Int -> A.Array Char -> t m Char
```

Comment:
This is very inefficient. You have all sorts of combinators available to you. Try to make use of them. Try not to use `toList` and `fromList`. Unless you're working explicitly with lists, you won't need to use them.

Reply:
What does `skpField` do? I think you can write it in a much simpler way.

Reply:
> What does `skpField` do? I think you can write it in a much simpler way.

It removes first `skpField opt` number of non-empty strings

Reply:
> This is very inefficient. You have all sorts of combinators available to you. Try to make use of them. Try not to use `toList` and `fromList`. Unless you're working explicitly with lists, you won't need to use them.

:+1: Used `toStream` to convert an Array to a Stream.
Using `S.concatMap`, I want to convert `t m String` to `t m Char`. Hence, I wanted to use an `S.fromList`.

Is it fine?

Reply:
```
{-# INLINE eqCh #-}

eqCh :: Char -> Char -> Bool
eqCh s c = (isSpace s == True && isSpace c == True) || (isSpace s == False && isSpace c == False)



skpField :: (IsStream t, Monad m) => Int -> A.Array Char -> t m Char
skpField n arr = S.concatMap S.fromList
               $ S.drop (2*n)
               $ S.groupsBy eqCh FL.toList
               $ S.cons ' '
               $ AI.toStream arr

-- dropping 2*n works only if there is at least one space in the beginning of a line


```

Reply:
Can you give me an example of what it should do? The input and the output.